tokio-reactor/src/lib.rs

```diff
@@ -392,23 +392,41 @@ impl Reactor {
         let aba_guard = token.0 & !MAX_SOURCES;
         let token = token.0 & MAX_SOURCES;
 
-        let io_dispatch = self.inner.io_dispatch.read().unwrap();
+        let mut rd = None;
+        let mut wr = None;
```

Comment:
tioli.. you could avoid the mut vars with

```rust
let (rd, wr) = {
    ...
    let wr = if ready.is_writable() || platform::is_hup(&ready) {
        io.writer.take_to_notify()
    } else {
        None
    };
    let rd = if !(ready & (!mio::Ready::writable())).is_empty() {
        io.reader.take_to_notify()
    } else {
        None
    };

    (rd, wr)
};
```

Reply:
I had that first, but I thought the mut vars was more readable 🤷‍♂️